### PR TITLE
feat(Templates): Updating Parameters Tab to Grid View to Align with Multi Workflow

### PR DESCRIPTION
**Pull request by Elaina-Lee**
## Requirement Checklist

* [X] The commit message follows our guidelines

## Type of Change

* [ ] Bug fix
* [X] Feature
* [ ] Other

Updating parameters tab to new design, aligning with multi workflow to display associated worklfow

## Screenshots or Videos (if applicable)
image
**Sorted by Name**
image
**Multi workflow (Showing Associated workflows)**
image
**After the tab has been moved to review+create and back (validation for the tab triggered)**
image

### CODE REVIEW

**Comment by Elaina-Lee** [comment]
5 mins - removing unwanted changes

File: libs/designer-ui/src/lib/flyout/index.tsx [hunk 1]
```diff
@@ -25,7 +26,7 @@ const defaultTooltipHostStyles: ITooltipHostStyles = {
 };
 
 export const Flyout = React.forwardRef<{ collapseFlyout(): void }, FlyoutProps>(
-  ({ ariaLabel, iconStyle, style, tabIndex = 0, text, tooltipHostStyles, onClick, title }, ref) => {
+  ({ ariaLabel, iconStyle, style, tabIndex = 0, text, tooltipHostStyles, onClick, title, iconSize = 'lg' }, ref) => {
```
Comment (preetriti1) [inline libs/designer-ui/src/lib/flyout/index.tsx:29]: was lg teh default size?

File: libs/logic-apps-shared/src/utils/src/lib/models/template.ts [hunk 2]
```diff
@@ -43,6 +43,7 @@ export interface Parameter {
 
 export interface ParameterDefinition extends Parameter {
   value?: any;
+  associatedWorkflows?: string[];
```
Comment (preetriti1) [inline libs/logic-apps-shared/src/utils/src/lib/models/template.ts:46]: Why do we have this list here? The manifest itself wont have any mention of this no?
We will have to determine based on the parameter key

File: libs/designer/src/lib/ui/templates/parameters/displayParameters.tsx [hunk 3]
```diff
@@ -1,37 +1,199 @@
...
+        onRenderItemColumn={onRenderItemColumn}
+        selectionMode={SelectionMode.none}
+      />
+    </div>
+  );
+};
+
+const copyAndSort = (items: Template.Parameter[], columnKey: string, isSortedDescending?: boolean): Template.Parameter[] => {
+  const keyPath =
+    columnKey === '$name' ? ['connectorDisplayName'] : columnKey === '$status' ? ['hasConnection'] : ['connection', 'displayName'];
```
Comment (preetriti1) [inline libs/designer/src/lib/ui/templates/parameters/displayParameters.tsx:189]: Pease change this.

File: libs/designer/src/lib/ui/templates/parameters/displayParameters.tsx [hunk 4]
```diff
@@ -1,37 +1,199 @@
...
+
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <div className="msla-templates-parameters-tab">
+      <DetailsList
+        setKey="id"
```
Comment (preetriti1) [inline libs/designer/src/lib/ui/templates/parameters/displayParameters.tsx:176]: Do you have id property in parameters?

**Review by preetriti1 — CHANGES_REQUESTED** [review]

File: libs/designer-ui/src/lib/flyout/index.tsx (hunk 1 above)
Comment (Elaina-Lee) [reply libs/designer-ui/src/lib/flyout/index.tsx:29]: The size that was being used was 28px, I wasn't sure what else to label this as we want something significantly smaller.

File: libs/logic-apps-shared/src/utils/src/lib/models/template.ts (hunk 2 above)
Comment (Elaina-Lee) [reply libs/logic-apps-shared/src/utils/src/lib/models/template.ts:46]: We could do that, but if we go that path, then everytime the parametersTab renders, we'd have to
   for every parameter
      for every manifest
         check if this parameter name exists, then added it to associatedWorkflows to display

I thought keeping it here as part of loading parameter process will be more runtime efficient

Comment (preetriti1) [reply libs/logic-apps-shared/src/utils/src/lib/models/template.ts:46]: We already have a way to do that.. from teh list of workflows in stores, we need to find each parameter exists in which workflow.parameterKeys... we dont have to look into manifest at all... i kept it that way for this reason

Comment (Elaina-Lee) [reply libs/logic-apps-shared/src/utils/src/lib/models/template.ts:46]: resolved it as discussed offline

File: libs/designer/src/lib/core/actions/bjsworkflow/templates.ts [hunk 5]
```diff
@@ -193,7 +192,25 @@ const loadTemplateFromResourcePath = async (templateName: string, manifest: Temp
         data.workflows[workflowPath] = workflowData.workflow;
         data.parameterDefinitions = {
           ...data.parameterDefinitions,
-          ...workflowData.parameterDefinitions,
+          ...Object.keys(workflowData.parameterDefinitions).reduce((acc: Record<string, Template.ParameterDefinition>, key: string) => {
```
Comment (preetriti1) [inline libs/designer/src/lib/core/actions/bjsworkflow/templates.ts:195]: I think this is going to run multiple times for shared parameters as well... so when you get the tests fixed, please do move this calculation after the for loop is done on the parameterDefinitions, so it only runs once for each unique parameter.

File: libs/designer/src/lib/ui/panel/templatePanel/__tests__/nameStateTab.spec.tsx [hunk 6]
```diff
@@ -22,10 +22,32 @@ describe('panel/templatePanel/createWorkflowPanel/nameStateTab', () => {
...
+            kinds: [],
+            artifacts: [
+              {
+                type: 'workflow',
+                file: 'workflow.json',
+              },
+              {
+                type: 'description',
+                file: 'description.md',
+              },
```
Comment (preetriti1) [inline libs/designer/src/lib/ui/panel/templatePanel/__tests__/nameStateTab.spec.tsx:41]: does this still exist?

**Review by preetriti1 — APPROVED** [review]

**Pull request merged**